pkg/xds/generator/prometheus_endpoint_generator_test.go

```diff
@@ -264,7 +274,7 @@ var _ = Describe("PrometheusEndpointGenerator", func() {
                 trafficDirection: INBOUND
                 address:
                   socketAddress:
-                    address: 0.0.0.0
+                    address: 192.168.0.1
```

Comment:
I am probably missing a seirous part of the idea behind this PR, but why we need to change the address here?

Reply:
First of all, it makes sense to expose this on `networking.address`, which is an address that DP is available on publicly. We didn't have this primitive back then when we were building metrics.
Second of all, `0.0.0.0` is potentially more conflicts. DirectAccessProxyGenerator would be more complicated.